sdk/compiler/damlc/daml-stdlib-src/DA/Crypto/Text.daml

```diff
@@ -78,14 +87,106 @@ class HasToHex a where
   -- | Converts a typed data value into a hex encoded string.
   toHex: a -> BytesHex
 
+instance HasToHex Text where
+  toHex = primitive @"BEEncodeHex"
+
+-- | HIDE
+intToHexByte: Int -> BytesHex
+intToHexByte n | 0 <= n && n <= 9 = show n
+intToHexByte 10 = "a"
+intToHexByte 11 = "b"
+intToHexByte 12 = "c"
+intToHexByte 13 = "d"
+intToHexByte 14 = "e"
+intToHexByte 15 = "f"
+intToHexByte n | n < 256 = (intToHexByte(n / 16)) <> (intToHexByte(n % 16))
+intToHexByte n = error $ "Int should be in range [0, 256): " <> show n
+
+instance HasToHex Int where
+  toHex n | 0 <= n && n <= 9 = "0" <> (show n)
+  toHex 10 = "0a"
+  toHex 11 = "0b"
+  toHex 12 = "0c"
+  toHex 13 = "0d"
+  toHex 14 = "0e"
+  toHex 15 = "0f"
+  toHex n | n < 256 = intToHexByte n
+  toHex n = (toHex(n / 256)) <> (toHex(n % 256))
```

Comment:
This repeated concatentation has quadratic cost in the lenght of the output. Apart from building a primitive, the better option is to build the list from right to left, and then call `Text.implode` (https://docs.daml.com/daml/stdlib/DA-Text.html#function-da-text-implode-82253)